test/unit-test/mqtt_agent_utest.c

```diff
@@ -1239,23 +1280,24 @@ void test_MQTTAgent_CommandLoop_add_acknowledgment_success( void )
 }
 
 /**
- * @brief Test that MQTTAgent_CommandLoop returns MQTTNoMemory if
- * pPendingAcks array is full.
+ * @brief Test that MQTTAgent_CommandLoop returns failure if an operation's entry
+ * cannot be added in the list of pending acknowledgments, either due to the array
+ * being full OR there being an existing entry for the same packet ID.
  */
 void test_MQTTAgent_CommandLoop_add_acknowledgment_failure( void )
 {
     MQTTStatus_t mqttStatus;
     MQTTAgentContext_t mqttAgentContext;
     size_t i = 0;
     Command_t commandToSend = { 0 };
+    uint16_t testPacketId = 1U;
```

Comment:
Could make this `const` if its value doesn't change
```suggestion
    const uint16_t testPacketId = 1U;
```